pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -610,30 +715,41 @@ func (gc *GarbageCollector) attemptToDeleteItem(ctx context.Context, item *node)
 			// otherwise, default to background.
 			policy = metav1.DeletePropagationBackground
 		}
-		klog.V(2).InfoS("Deleting object", "object", klog.KRef(item.identity.Namespace, item.identity.Name),
-			"objectUID", item.identity.UID, "kind", item.identity.Kind, "propagationPolicy", policy)
+		logger.V(2).Info("Deleting object",
+			"object", klog.KRef(item.identity.Namespace, item.identity.Name),
+			"objectUID", item.identity.UID,
+			"kind", item.identity.Kind,
+			"virtual", !item.isObserved(),
+			"propagationPolicy", policy,
+		)
 		return gc.deleteObject(item.identity, &policy)
 	}
 }
 
 // process item that's waiting for its dependents to be deleted
-func (gc *GarbageCollector) processDeletingDependentsItem(item *node) error {
+func (gc *GarbageCollector) processDeletingDependentsItem(ctx context.Context, item *node) error {
+	logger := klog.FromContext(ctx)
+	logger = klog.LoggerWithValues(logger, "item", item.identity)
+
 	blockingDependents := item.blockingDependents()
 	if len(blockingDependents) == 0 {
-		klog.V(2).Infof("remove DeleteDependents finalizer for item %s", item.identity)
-		return gc.removeFinalizer(item, metav1.FinalizerDeleteDependents)
+		logger.V(2).Info("remove DeleteDependents finalizer for item", "item", item.identity)
+		return gc.removeFinalizer(ctx, item, metav1.FinalizerDeleteDependents)
```

Comment:
There's a slight bug here: `logger` was updated, but the `ctx` still uses the old logger.

I think all of these `gc` methods should better take a `logger` instead of `ctx`, then the need to create a new context with the updated logger goes away and those functions don't need to call `FromContext`.

Reply:
Some of the gc methods already take in a context (e.g. for making a client call). Would you rather they become more uniform (take in a context when they didn't before), or have a mix of taking in a logger (new addition) or a context (existing code)?

Reply:
I think a mix is fine. It's always a case-by-case decision.